src/authorized_keys/authorized_keys.cc

```diff
@@ -12,85 +12,69 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <cstdlib>
 #include <iostream>
-#include <sstream>
-#include <string>
 
 #include <signal.h>
 
 #include <oslogin_utils.h>
 
 using std::cout;
 using std::endl;
-using std::string;
 
-using oslogin_utils::HttpGet;
-using oslogin_utils::ParseJsonToSuccess;
-using oslogin_utils::ParseJsonToEmail;
+using oslogin_utils::AuthOptions;
+using oslogin_utils::AuthorizeUser;
+using oslogin_utils::CloseSysLog;
 using oslogin_utils::ParseJsonToSshKeys;
-using oslogin_utils::UrlEncode;
-using oslogin_utils::kMetadataServerUrl;
+using oslogin_utils::SetupSysLog;
+using oslogin_utils::SysLogErr;
 
-void sigpipe_handler(int signo) {
-  // exit 0 so SSHD can use what we've already written out.
+#define SYSLOG_IDENT "sshd"
+#define SYSLOG_PREFIX "google_authorized_keys"
+#define SUCCESS 0
+#define FAIL    1
+
+void signal_handler(int signo) {
   _Exit(0);
 }
 
-int main(int argc, char* argv[]) {
+int main(int argc, char *argv[]) {
+  struct AuthOptions opts;
+  struct sigaction sig;
+  char *user_name;
+  string user_response;
+
+  SetupSysLog(SYSLOG_IDENT, SYSLOG_PREFIX);
+
   if (argc != 2) {
-    cout << "usage: authorized_keys [username]" << endl;
-    return 1;
+    SysLogErr("usage: authorized_keys [username]");
```

Comment:
why not argv[0]?

Reply:
do you mean `SysLogErr("usage: %s [username]", argv[0])` ?

Reply:
Done. Thanks.